### DISCUSSION TITLE
Opt out CSRF per route

### DISCUSSION THREAD

**Discussion by florian-lefebvre** [withastro/roadmap#923] [Proposal]
### Body

# Summary

Allow to opt out of CSRF protection per route.

# Background & Motivation

While it's good practice to have CSRF enable for an entire Astro project, there are cases where not having CSRF enabled for a specific route is required. One example is a webhook, [read Stripe docs](https://docs.stripe.com/webhooks#csrf-protection).

# Goals

- Having a way to opt-out CSRF protection for a given route

# Example

```ts
export const checkOrigin = false;
```

The risk of this API is to have too many per-route setting (like `prerender`) so it may need a RFC before to to tackle this topic ([discord thread](https://discord.com/channels/830184174198718474/1214261710815367232))

**Comment by mseibert** [comment]
We would need this in our projects.

**Reply by florian-lefebvre** [reply]
I don't think this is likely to move anytime soon FYI. Ping me on the related private issue so we can discuss alternatives

**Reply by hcjmartin** [reply]
I'd agree this would have been nice to have a simple exclusion config list rather than re-implementing the same middleware. 

Any particular reason why this should be avoided? Otherwise happy to submit for PR.

**Reply by florian-lefebvre** [reply]
I think with https://github.com/withastro/roadmap/pull/1344 there's the potential to make CSRF just another middleware with more control

cc @matthewp

**Reply by matthewp** [reply]
Yeah good idea, I hadn't thought about spinning this out as its own handler/middleware but that makes sense to do.
